homeassistant/components/fritz/common.py

```diff
@@ -495,6 +455,42 @@ def _async_remove_empty_devices(
                 _LOGGER.info("Removing device: %s", device_entry.name)
                 device_reg.async_remove_device(device_entry.id)
 
+    async def service_fritzbox(
+        self, service_call: ServiceCall, config_entry: ConfigEntry
+    ) -> None:
+        """Define FRITZ!Box services."""
+        _LOGGER.debug("FRITZ!Box service: %s", service_call.service)
+
+        if not self.connection:
+            raise HomeAssistantError("Unable to establish a connection")
+
+        try:
+            if service_call.service == SERVICE_REBOOT:
+                _LOGGER.warning(
+                    'Service "fritz.reboot" is deprecated, please use the corresponding button entity instead'
+                )
+                await self.async_trigger_reboot()
+                return
+
+            if service_call.service == SERVICE_RECONNECT:
+                _LOGGER.warning(
+                    'Service "fritz.reconnect" is deprecated, please use the corresponding button entity instead'
+                )
+                await self.async_trigger_reconnect()
+                return
+
+            if service_call.service == SERVICE_CLEANUP:
+                _LOGGER.warning(
+                    'Service "fritz.cleanup" is deprecated, please use the corresponding button entity instead'
+                )
+                await self.async_trigger_cleanup(config_entry)
+                return
+
+        except (FritzServiceError, FritzActionError) as ex:
+            raise HomeAssistantError("Service or parameter unknown") from ex
+        except FritzConnectionException as ex:
+            raise HomeAssistantError("Service not supported") from ex
+
```

Comment:
should be moved into `service.py` 🤔

Reply:
not worth the rework as services will be removed in a near future ;-)